api/authentication.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework import exceptions
+from rest_framework.authentication import TokenAuthentication, get_authorization_header
+from api.models import Token, IP
+
+
+class TokenIPAuth(TokenAuthentication):
```

Comment:
Double check if this really needs to be a subclass of TokenAuthentication. It's re-implementing everything and also overriding the interface for some methods as well.